api/v3/Job/UpdateCHContributions.php

```diff
@@ -28,6 +28,10 @@ function _civicrm_api3_job_update_c_h_contributions_spec(&$spec) {
 function civicrm_api3_job_update_c_h_contributions($params) {
   $batchSize = CRM_Utils_Array::value('batch_size', $params, 1000);
   $dao = CRM_Core_DAO::executeQuery("SELECT * FROM civicrm_ch_contribution_batch LIMIT 0, $batchSize ");
+  $lock = Civi::lockManager()->acquire("data.core.update.chcontribution." . time());
```

Comment:
actually @monishdeb won't this mean a new lock is created on every cron run? do we need to have the lock more like how the mailing job does it on each contribution i.e. move it within the dao->fetch call? https://github.com/civicrm/civicrm-core/blob/master/CRM/Mailing/BAO/MailingJob.php#L118